test/parallel/test-stringbytes-external-at-max.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+// v8 fails silently if string length > v8::String::kMaxLength
+(function() {
+  // v8::String::kMaxLength defined in v8.h
+  const kStringMaxLength = process.binding('buffer').kStringMaxLength;
+
+  try {
+    new Buffer(kStringMaxLength * 3);
```

Comment:
The reason for the over allocation is to also judge the amount of extra space the heap has available. If we only allocate enough for the test then the `toString()` operations may make v8 abort because of unavailable memory.